src/osemgrep/language_server/Test_LS_e2e.ml

```diff
@@ -144,7 +121,16 @@ let open_and_write_default_content ?(mode = []) file =
    how the promises resolve, such that the sleeping thread would stay
    sleeping and the entire process would eventually exit.
    So, when running in JSCaml, let's just not use pauses. *)
-let lwt_pause () = if !Common.jsoo then Lwt.return_unit else Lwt.pause ()
+let lwt_pause () =
+  if !Common.jsoo then Lwt.return_unit
+  else
+    (* For some reason, Lwt.pause makes the Sys.cwd change.
+       I have no idea why. But this fixes that.
+    *)
+    let dir = Sys.getcwd () in
+    let%lwt () = Lwt.pause () in
+    Sys.chdir dir;
```

Comment:
Why sometimes using Sys.chdir and sometimes Unix.chdir?

Reply:
See the added comment

Reply:
Something in another lwt promise calls chdir without restoring it. What the magic call does here is schedule the function `fun () -> Sys.chdir dir; Lwt.return_unit` to be called when the lwt scheduler sees fit. It's not `Lwt.pause ()` that calls chdir but some other task that's executing before our own. This is bad and must be fixed at the source (i.e. not here although the workaround is correct) by a synchronous `with_chdir dir (fun () -> ...)` that takes care of restoring the original current directory even in case of an exception.